resip/recon/ConversationManager.hxx

```diff
@@ -483,6 +486,18 @@ public:
    virtual void onDtmfEvent(ParticipantHandle partHandle, int dtmf, int duration, bool up) = 0;
 
    ///////////////////////////////////////////////////////////////////////
+   // Pager Message Handlers /////////////////////////////////////////////
+   ///////////////////////////////////////////////////////////////////////
+
+   virtual void onMessageArrived(resip::ServerPagerMessageHandle handle, const resip::SipMessage& message) = 0;
```

Comment:
The InviteSessionHandler onMessage is for in-dialog MESSAGE requests only
(ie:  MESSAGE requests received in the context of an existing INVITE
session).  The standard for IM's is to use out-of-dialog MESSAGE requests,
so you may want to handle both, but definitely want to handle at least
using the Pager classes.

Scott

On Tue, Aug 9, 2016 at 2:21 PM, Mateus Augusto Bellomo Agrello Ruivo <
notifications@github.com> wrote:

> In resip/recon/ConversationManager.hxx
> <https://github.com/resiprocate/resiprocate/pull/53#discussion_r74115617>:
>
> > @@ -483,6 +486,18 @@ public:
> >     virtual void onDtmfEvent(ParticipantHandle partHandle, int dtmf, int duration, bool up) = 0;
> >
> >     ///////////////////////////////////////////////////////////////////////
> > +   // Pager Message Handlers /////////////////////////////////////////////
> > +   ///////////////////////////////////////////////////////////////////////
> > +
> > +   virtual void onMessageArrived(resip::ServerPagerMessageHandle handle, const resip::SipMessage& message) = 0;
>
> The method onMessageArrived() is called at [1] using the
> ServerPagerMessageHandler that was set at resip/recon/UserAgent.cxx:
>
> mDum.setServerPagerMessageHandler(mConversationManager);
>
> The method onMessage() is coming from InviteSessionHandler that
> ConversationManager derived from. Do you think I could use this same method
> for incoming text messages?
>
> [1] https://github.com/resiprocate/resiprocate/blob/master/resip/dum/
> ServerPagerMessage.cxx#L86
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/resiprocate/resiprocate/pull/53/files/9e3b4ac1410cb4a880dfbe7a68727478b961d5c0#r74115617>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACtNUV7Q6m7YwbGGmUtmwpl0mjo6_TFwks5qeMU3gaJpZM4JYkAY>
> .
>